packages/wonder-blocks-birthday-picker/src/components/birthday-picker.js

```diff
@@ -0,0 +1,236 @@
+// @flow
+import moment from "moment";
+import * as React from "react";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {Body} from "@khanacademy/wonder-blocks-typography";
+import Icon, {icons} from "@khanacademy/wonder-blocks-icon";
+import Color from "@khanacademy/wonder-blocks-color";
+import {SingleSelect, OptionItem} from "@khanacademy/wonder-blocks-dropdown";
+
+type Props = {|
+    /**
+     * The default value to populate the birthdate with. Should be in the
+     * format: YYYY-MM-DD (e.g. 2021-05-26). It's only used to populate the
+     * initial value as this is an uncontrolled component.
+     */
+    defaultValue?: string,
+
+    /**
+     * Listen for changes to the birthdate. Could be a string in the YYYY-MM-DD
+     * format or `null`.
+     */
+    onChange: (date: ?string) => mixed,
+|};
+
+type State = {|
+    month: string | null,
+    day: string | null,
+    year: string | null,
+    error: string | null,
+|};
+
+// Flow doesn't know about the getYear property on Date for some reason!
+// $FlowFixMe[prop-missing]
+const CUR_YEAR = new Date().getYear() + 1900;
+// TODO(WB-1200.2): Add labels prop to handle translations.
+const BIRTHDAY_ERROR = "Please select a valid birthdate.";
+
+/**
+ * Birthday Picker. Similar to a datepicker, but specifically for birthdates.
+ * We don't want to show a calendar in this case as it can be quite tedious to
+ * try and select a date that's many years old. Instead, we use a set of
+ * dropdowns to achieve a similar effect.
+ *
+ * More information on this pattern:
+ * https://medium.com/samsung-internet-dev/making-input-type-date-complicated-a544fd27c45a
+ *
+ * Arguably, this should probably even be 3 textfields, but that would be a
+ * larger design change, more info:
+ * https://designnotes.blog.gov.uk/2013/12/05/asking-for-a-date-of-birth/
+ *
+ * **NOTE:** This component is uncontrolled.
+ *
+ * ### Usage
+ *
+ * ```jsx
+ * import {BirthdayPicker} from "@khanacademy/wonder-blocks-dates";
+ *
+ * <BirthdayPicker
+ *  defaultValue="2021-05-26"
+ *  onChange={(date) => {setDate(date)}}
+ * />
+ * ```
+ */
+export default class BirthdayPicker extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        let month = null;
+        let day = null;
+        let year = null;
+        let error = null;
```

Comment:
I find `let` to be a smell that we need a separate function. What do you think about a `getStateFromDefault` call that returned a state object?